cmd/kubeadm/app/cmd/phases/reset/unmount_linux.go

```diff
@@ -0,0 +1,47 @@
+// +build linux
+
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package phases
+
+import (
+	"io/ioutil"
+	"strings"
+	"syscall"
+
+	"k8s.io/klog"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+)
+
+// unmountKubeletDirectory unmounts all paths that contain KubeletRunDirectory
+func unmountKubeletDirectory() error {
+	raw, err := ioutil.ReadFile("/proc/mounts")
+	if err != nil {
+		return err
+	}
+	mounts := strings.Split(string(raw), "\n")
+	for _, mount := range mounts {
+		m := strings.Split(mount, " ")
+		if len(m) < 2 || !strings.HasPrefix(m[1], kubeadmconstants.KubeletRunDirectory) {
+			continue
+		}
+		if err := syscall.Unmount(m[1], syscall.MNT_FORCE); err != nil {
```

Comment:
I can remove this if you like. will have to be tonight as it will mean I need to retest it.

Reply:
although i'm confused as everything I read about `force` implies that it will `force unmount (in case of an unreachable NFS system)` (from the command line)
https://golang.org/src/syscall/syscall_linux.go?s=20289:20384#L979
makes it look like umount(2) is used. who has a force flag for the above reason.

digging though the code used on ubuntu 18.04
`_, _, e1 := Syscall(SYS_UMOUNT2, uintptr(unsafe.Pointer(_p0)), uintptr(flags), 0)`  would seem to support that. and unmount to `umount2` and `MNT_FORCE` to `umount2` is defined as 
"MNT_FORCE (since Linux 2.1.116)
    Force unmount even if busy. This can cause data loss. (Only for NFS mounts.) " from https://linux.die.net/man/2/umount

Anyhow. maybe i'm missing something. please let me know! Thanks for your insight!

Reply:
this has been removed and tested. (testing methodology posted.) I agree, it's misleading and really not helpful in this case. Also not used on the original so even if we wanted to add it would probably be better to add in new PR.